checks/check-value-required.ts

```diff
@@ -0,0 +1,39 @@
+import { Zero } from '@ethersproject/constants'
+import { BigNumber } from '@ethersproject/bignumber'
+import { formatEther } from '@ethersproject/units'
+import { ProposalCheck } from '../types'
+
+/**
+ * Reports on whether the caller initiating the `execute` call needs to send ETH with the call.
+ */
+export const checkValueRequired: ProposalCheck = {
+  name: 'Reports on whether the caller needs to send ETH with the call',
+  async checkProposal(proposal, sim, deps) {
+    // TODO Fix typings for values. The `values` field is not always present in the proposal object,
+    // but key `3` contains them. (Similarly key 0 is proposal ID, 1 is proposer, etc.). This is
+    // related to why we use `proposalCreatedEvent.args![3]` in `tenderly.ts`.
+    type ProposalValues = { '3': BigNumber[] }
+    const totalValue = proposal.values
+      ? // For local simulations, `values` exists and `3` does not.
+        proposal.values.reduce((sum, cur) => sum.add(cur), Zero)
+      : // For simulations read from the chain, `3` exists and `values` does not.
+        (proposal as unknown as ProposalValues)['3'].reduce((sum, cur) => sum.add(cur), Zero)
+
+    const txValue = BigNumber.from(sim.simulation.value)
+    if (txValue.eq(Zero)) {
+      const msg = 'No ETH is required to be sent by the account that executes this proposal.'
+      return { info: [msg], warnings: [], errors: [] }
+    }
+
+    const valueRequired = formatEther(totalValue)
+    const govBalance = formatEther(await deps.provider.getBalance(deps.governor.address))
+    const valueSent = formatEther(txValue)
```

Comment:
Possibly a dumb question, but why don't we compare `valueRequired` to `valueSent` and return an error if the required amount is less than what was sent?

Reply:
I guess there can be a case where one of the call will increase the timelock balance to fund the subsequent call.